huxley/api/tests/__init__.py

```diff
@@ -107,6 +115,11 @@ class CreateAPITestCase(AbstractAPITestCase):
     is_resource = False
 
 
+class PostSingleAPITestCase(AbstractAPITestCase):
```

Comment:
Rename this to Create...
Also, why do you need is_resource to be true?

Reply:
In `get_url` method in **tests/__init__.py**

`args = (object_id, ) if self.is_resource else ()`
`return reverse(self.url_name, args=args)`

So `is_resource` must be true or it doesn't generate the url correctly

If you look at **urls.py** the urls for committee_feedback are:

`    url(r'^committee_feedback/?$',`
`        views.committee_feedback.CommitteeFeedbackList.as_view(),`
`        name='committee_feedback_list'),`
`    url(r'^committee_feedback/(?P<committee>[0-9]+)/?$',`
`        views.committee_feedback.CommitteeFeedbackDetail.as_view(),`
`        name='committee_feedback_detail'),`

So you need to have `is_resource` be true so the post request goes to the committee_feedback_detail url.

Now that I'm looking at this again, do you think this needs to be changed?

There's no way to reference a single committee feedback right now. For this feature, that isn't an issue because we only ever retrieve them in a lists for the ChairCommitteeFeedbackView. However if we wanted to add something else that requires being able to reference a single feedback, we don't have that right now.

If we do change it, how do we go to that url to create a feedback without knowing the feedback's ID (Because it hasn't been created in the database)? Could you just pass in a random integer?

Other than that one question, I have no issue changing this and I don't think it would be too much work. Also, it would be more consistent with how the rest of the api works.

Reply:
Okay I messed around with that kind of implementation on my machine and it works completely fine. You just have to pass in a random integer.

Maybe we could set it so that all post requests for single entry use 0 for the pk. All SQL databases start autoincrementing at 1 and all other HTTP requests would access an entry in the db that already exists.

Reply:
I've got a simpler way around this: change the detail url to be a fixed address (say committee_feedback/post). There's actually never a reason to pass pk to the url, since we never use this url to do database lookups (and likely never will).